### PR TITLE
Cannot add mentor to internship engagement

### PR DESCRIPTION
**Pull request by frankyma**
The old update mentor only worked a mentor already existed because it used a hard match.  Now using optional match to set the old to inactive.

Same pattern for location

### CODE REVIEW

File: test/engagement.e2e-spec.ts [hunk 1]
```diff
@@ -357,14 +358,18 @@ describe('Engagement e2e', () => {
     expect(updated.status).toBe(EngagementStatus.Rejected);
   });
 
-  it('updates internship engagement', async () => {
+  it.only('updates internship engagement', async () => {
```
Comment (CarsonF) [inline test/engagement.e2e-spec.ts:361]:
```suggestion
  it('updates internship engagement', async () => {
```

Comment (frankyma) [reply test/engagement.e2e-spec.ts]: ok making a mental note for myself to always double check this 😬

Comment (CarsonF) [reply test/engagement.e2e-spec.ts]: We need a lint rule lol

Comment (frankyma) [reply test/engagement.e2e-spec.ts]: There's a few in the market. https://www.npmjs.com/package/eslint-plugin-no-only-tests?

**Review by michaelmarshall — APPROVED** [review]

**Pull request merged**